src/common/utils.ts

```diff
@@ -169,7 +169,7 @@ function isHex(hex: string): boolean {
 }
 
 function cleanHex(hexWithMaybePrefix: string): string {
-  return hexWithMaybePrefix.startsWith('0x')
+  return typeof hexWithMaybePrefix === 'string' && hexWithMaybePrefix.startsWith('0x')
```

Comment:
This shouldn't ever be necessary with TypeScript; we're already declaring the type. 

The problem lies with why a buffer is being passed to `cleanHex`, and not causing a compile-time type error in the first place.